src/esp/assets/ResourceManager.cpp

```diff
@@ -1034,43 +1013,31 @@ void ResourceManager::loadTextures(Importer& importer, MeshMetaData* metaData) {
 //! Add component to rendering stack, based on importer loading
 //! TODO (JH): decouple importer part, so that objects can be
 //! instantiated any time after initial loading
-void ResourceManager::addComponent(Importer& importer,
-                                   const MeshMetaData& metaData,
+void ResourceManager::addComponent(const MeshMetaData& metaData,
                                    scene::SceneNode& parent,
                                    DrawableGroup* drawables,
-                                   int componentID) {
-  std::unique_ptr<Magnum::Trade::ObjectData3D> objectData =
-      importer.object3D(componentID);
-  if (!objectData) {
-    LOG(ERROR) << "Cannot import object " << importer.object3DName(componentID)
-               << ", skipping";
-    return;
-  }
-
+                                   MeshTransformNode& meshTransformNode) {
```

Comment:
```suggestion
                                   const MeshTransformNode& meshTransformNode) {
```

I don't see why this should be a mutable reference.